### ISSUE TITLE
each-in breaks for getters and keys with dots

### ISSUE THREAD

**Issue by asakusuma** [emberjs/ember.js#17927]
1. If you pass an object defined with `defineProperties`, the property is ignored.
2. If you pass an object with a property that has a key with a dot in it, the property value is empty.

Whatever version ember-twiddle uses (3.4.3 I think) doesn't have the dot key problem, but it does have the getter problem: https://ember-twiddle.com/306025f713fe62a25bf8eb61dafc5e1d

Reproduction on 3.9 that has both problems: https://github.com/asakusuma/each-in-getter-repro/commit/bd2dab369a4369c95bfd4af2ee2db46678eae636

**Comment by rwjblue** [comment]
Duplicate of https://github.com/emberjs/ember.js/issues/17529, would you mind chiming in over there?

**Comment by asakusuma** [comment]
@rwjblue I believe the `If you pass an object defined with defineProperties, the property is ignored.` is not a duplicate. I think it's a separate issue. The twiddle is on ember 3.4, and can reproduce the `defineProperties` issue. I believe https://github.com/emberjs/ember.js/issues/17529 was introduced much later.

**Comment by rwjblue** [comment]
The second item (the one that is talking about `{{#each-in` with an object with keys that have a period in them) is working now, and I've debugged to figure out what is going on in the other case. 

`{{#each-in` is iterating over all of a given object's keys, effectively `Object.keys(theObject)`. If we were to use `{ get foo() { return 'bar'; } }` in the demo, it works perfectly. The difference is that in the `defineProperties` version, we have not declared that `foo` is **enumerable**!.

**tldr;** Changing your demo to:

```js
    const getter = Object.create(Object.prototype, {
      foo: {
        enumerable: true,
        get: function () {
          return 'bar';
        }
      }
    });
```

Works perfectly 😸.
